packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -263,7 +272,7 @@ class DecorationImagePainter {
       canvas: canvas,
       rect: rect,
       image: _image.image,
-      scale: _image.scale,
+      scale: _details.scale,
```

Comment:
It seems odd to me that we now completely ignore the scale of the actual image.

/cc @dnfield

Reply:
This is almost certainly wrong, and unfortunately it doesn't look like we have a test covering it.

I would expect this to give you strange results if you now tried to use a `DecorationImage` with an `AssetImage` of a non-1.0 scale.

Reply:
@dnfield @goderbauer I understand the issue with this approach. It completely ignores the scale given during `NetworkImage` and `ExactAssetImage`. 

But, if we are using scale property of `ImageInfo` instead of `DecorationImage`, we will have to create a new `ImageProvider` before creating an `ImageStream`, which means we would have to override the scale of `ImageInfo` with `DecorationImage`, which we are doing in the current approach. 

What if we document this behaviour in the API docs?

Reply:
I think if we just multiplied the scale property from the decoration against the image scale, it should work out.

Reply:
We should also add some tests that show that that works ok with and without this parameter.

Reply:
And we should document this behavior - someone could then override it by wrapping the incoming provider with a new provider that sets the scale to 1.0.

Reply:
@dnfield I'll do the required changes. I'll add the test which shows that it works properly with `scale` property. We already have test cases which don't use `scale` property, [example](https://github.com/flutter/flutter/blob/master/packages/flutter/test/painting/decoration_test.dart#L223)